scipy/stats/tests/test_multivariate.py

```diff
@@ -1974,6 +1975,55 @@ def test_to_corr(self):
         assert_allclose(m[0,0], 1)
 
 
+class TestRandomDirection:
+    @pytest.mark.parametrize("dim", [1, 3])
+    @pytest.mark.parametrize("size", [None, 5, (5, 4)])
+    def test_samples(self, dim, size):
+        # test that samples have correct shape and norm 1
+        random_direction_dist = random_direction(dim)
+        samples = random_direction_dist.rvs(size)
+        if isinstance(size, int):
+            size = (size, )
+        elif size is None:
+            size = (1, )
+        expected_shape = size + (dim, )
+        assert samples.shape == expected_shape
+        norms = np.linalg.norm(samples, axis=-1)
+        assert_allclose(norms, 1.)
+
+    @pytest.mark.parametrize("dim", [None, 0, (2, 2), 2.5])
+    def test_invalid_dim(self, dim):
+        message = ("Dimension of vector must be specified, "
+                   "and must be an integer greater than 0.")
+        with pytest.raises(ValueError, match=message):
+            random_direction.rvs(dim)
+
+    def test_frozen_distribution(self):
+        dim = 5
+        frozen = random_direction(dim)
+        frozen_seed = random_direction(dim, seed=514)
+
+        rvs1 = frozen.rvs(random_state=514)
+        rvs2 = random_direction.rvs(dim, random_state=514)
+        rvs3 = frozen_seed.rvs()
+
+        assert_equal(rvs1, rvs2)
+        assert_equal(rvs1, rvs3)
+
+    @pytest.mark.parametrize("dim", [2, 5, 8])
+    def test_uniform(self, dim):
+        spherical_dist = random_direction(dim)
+        samples = spherical_dist.rvs(size=10000, random_state=42967295)
+        angles = np.arctan2(samples[:, dim - 1], samples[:, dim - 2])
+        # test that angles follow a uniform distribution
+        # normalize angles to range [0, 1]
+        angles += np.pi
+        angles /= 2*np.pi
```

Comment:
```suggestion
        rng = np.random.default_rng(1036978481269651776)
        spherical_dist = random_direction(dim, seed=rng)
        # generate random, orthogonal vectors
        v1, v2 = spherical_dist.rvs(size=2)
        v2 -= v1 @ v2 * v1
        v2 /= np.linalg.norm(v2)
        assert_allclose(v1 @ v2, 0, atol=1e-14)  # orthogonal
        # generate data and project onto orthogonal vectors
        samples = spherical_dist.rvs(size=10000)
        s1 = samples @ v1
        s2 = samples @ v2
        angles = np.arctan2(s1, s2)
```
This is a bit more general to show that we're not cheating. (If someone could check my math, I'd appreciate it.)